dotnet/src/Connectors/Connectors.MistralAI.UnitTests/Client/MistralClientTests.cs

```diff
@@ -238,117 +246,236 @@ await foreach (var chunk in response)
-            {
-                this.ResponseMessage = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
-                {
-                    Content = new StreamContent(new MemoryStream(this._responseBytes))
-                };
-            }
-
-            return await Task.FromResult(this.ResponseMessage);
-        }
+            invokedFunctions.Add(context.Function.Name);
+            await next(context);
+            context.Terminate = true;
+        });
+        kernel.AutoFunctionInvocationFilters.Add(filter);
+
+        // Act
+        var executionSettings = new MistralAIPromptExecutionSettings { ToolCallBehavior = MistralAIToolCallBehavior.AutoInvokeKernelFunctions };
+        var chatHistory = new ChatHistory
+        {
+            new ChatMessageContent(AuthorRole.User, "What is the weather like in Paris?")
+        };
+        var response = await client.GetChatMessageContentsAsync(chatHistory, default, executionSettings, kernel);
+
+        // Assert
+        Assert.NotNull(response);
+        Assert.Single(response);
+        Assert.Equal("12°C\nWind: 11 KMPH\nHumidity: 48%\nMostly cloudy", response[0].Content);
+        Assert.Null(response[0].ModelId);
+        Assert.Equal(1, this.DelegatingHandler.SendAsyncCallCount);
+        Assert.Equal(3, chatHistory.Count);
+        Assert.Contains("GetWeather", invokedFunctions);
     }
 
+    #region internal classes
     public sealed class WeatherPlugin
     {
         [KernelFunction]
         [Description("Get the current weather in a given location.")]
         public string GetWeather(
-            [Description("The city and department, e.g. Marseille, 13")] string location,
-            [Description("The temperature units one of celsius or fahrenheit")] TemperatureUnit units
-            ) => $"{{\"location\": \"{location}\", \"unit\": \"{units}\"}}";
+            [Description("The city and department, e.g. Marseille, 13")] string location
+            ) => "12°C\nWind: 11 KMPH\nHumidity: 48%\nMostly cloudy";
+    }
+
+    internal enum TemperatureUnit { Celsius, Fahrenheit }
+
+    public class WidgetFactory
+    {
+        [KernelFunction]
+        [Description("Creates a new widget of the specified type and colors")]
+        public string CreateWidget([Description("The colors of the widget to be created")] WidgetColor[] widgetColors)
+        {
+            var colors = string.Join('-', widgetColors.Select(c => c.GetDisplayName()).ToArray());
+            return $"Widget created with colors: {colors}";
+        }
+    }
+
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public enum WidgetColor
+    {
+        [Description("Use when creating a red item.")]
+        Red,
+
+        [Description("Use when creating a green item.")]
+        Green,
+
+        [Description("Use when creating a blue item.")]
+        Blue
     }
 
-    public enum TemperatureUnit { Celsius, Fahrenheit }
+    internal sealed class FakeFunctionFilter(
```

Comment:
I assume you are not using this subclass outside of this test, make all private.

```suggestion
    private sealed class FakeFunctionFilter(
```